presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -169,8 +170,14 @@ public Page getNextPage()
                     blocks[fieldId] = constantBlocks[fieldId].getRegion(0, batchSize);
                 }
                 else {
-                    int fieldIndex = requestedSchema.getFieldIndex(columnNames.get(fieldId));
```

Comment:
This is the line where the new reader always relies on column names and was causing issue with a table where a column was renamed.